djangocms_versioning/cms_toolbars.py

```diff
@@ -86,7 +86,21 @@ def _add_edit_button(self, disabled=False):
             )
             self.toolbar.add_item(item)
 
+    def _remove_cms_edit_button(self):
+        """
+        Find and remove the edit edit button with the CMS.
+        """
+        # Loop through each button container
+        for button_list in self.toolbar.get_right_items():
+
+            # Try and locate an edit button
+            found = [button for button in button_list.buttons if button.name == _('Edit')]
+            if found:
+                button_list.buttons.remove(found[0])
+                return
+
     def post_template_populate(self):
         super(VersioningToolbar, self).post_template_populate()
+        self._remove_cms_edit_button()
```

Comment:
I don't think you will need to iterate button items and remove the Edit button. With the new implementation in CMS 4.0.x, you can extend VersioningToolbar with PlaceholderToolbar and override `add_edit_button`.

Reply:
@malindap92 I couldn't remember the plan for this, extending the Placeholder forces a page object which felt wrong, although I agree that targeting by the label "edit" is poor. Can you remember if the plan was to extend the Placeholder toolbar?

Reply:
Earlier the buttons were rendered by PageToolbar. This has been moved to PlaceholderToolbar and its implemented in a way where you can extend and override `add_edit_button` and `add_preview_button`. Please do clarify with @chaosk as well.

Reply:
@malindap92 Soo, we need to inherit from PlaceholderToolbar, unregister it and register our toolbar, right? I think @Aiky30 raised a point that some other apps could be trying to do the same, and once PlaceholderToolbar is unregistered they wouldn't be able to find ours.

Reply:
@chaosk Yes that was my assumption that VersioningToolbar will override PlaceholderToolbar. In ModerationToolbar, it inherits VersioningToolbar and unregister it. Version Locking monkey patches the VersioningToolbar. Please do point me if I'm missing something here.

Reply:
If the buttons where removed from the PageToolbar to facilitate overriding the PlaceholderToolbar we should go with that rather than monkey patching or as I've done look by the button name.